scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -1156,6 +1156,13 @@ impl ProcMacroHost {
         Ok(())
     }
 
+    /// Registers a prebuilt procedural macro by loading the shared library from the specified path.
+    pub fn register_prebuilt(&mut self, package: Package, config: &Config) -> Result<()> {
```

Comment:
I don't see why this needs to be a separate function. Can't `ProcMacroHost::register` decide which logic should be used?

Reply:
I assume what you mean is `ProcMacroHost::register` should be making the decision on loading mechanism implicitly based on the package being in the allow-list and prebuilt binary being present.

> When loading procmacro package ~that is present in the allowlist~ (#1769), Scarb will look for the `/target/scarb/cairo-plugin` directory and will attempt to load a matching binary if present. In any case this loading process fails, Scarb will always fall back to source-code compilation. This will happen silently without any warnings.

However, we want to be able to attempt loading from prebuilts, and if that resulted in failure for any reason, explicitly compile and load the binaries instead.